### PR TITLE
coreterminal: init at 4.2.0

### PR DESCRIPTION
**Pull request by dan4ik605743**
###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [x] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/terminal-emulators/coreterminal/qtermwidget/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  buildInputs = [
+    qt5.full
+    lxqt.lxqt-build-tools
+    python3Packages.pyqt5
+    python3Packages.sip
+    python3Packages.sip_4
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/coreterminal/qtermwidget/default.nix:30]: Why do we need both versions?

File: pkgs/applications/terminal-emulators/coreterminal/default.nix [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+, fetchzip
+, callPackage
+, cmake
+, ninja
+}:
+
+let
+  qtermwidget = callPackage ./qtermwidget/default.nix {};
+  libcprime = callPackage ./libcprime/default.nix {};
+in
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/coreterminal/default.nix:12]: Move those to all packages.

File: pkgs/applications/terminal-emulators/coreterminal/default.nix [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+, lib
+, fetchzip
+, callPackage
+, cmake
+, ninja
+}:
+
+let
+  qtermwidget = callPackage ./qtermwidget/default.nix {};
+  libcprime = callPackage ./libcprime/default.nix {};
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/coreterminal/default.nix:11]:
```suggestion
  libcprime = libsForQt5.callPackage ./libcprime { };
```

File: pkgs/applications/terminal-emulators/coreterminal/default.nix [hunk 4]
```diff
@@ -0,0 +1,41 @@
+{ qt5
+, lib
+, fetchzip
+, callPackage
+, cmake
+, ninja
+}:
+
+let
+  qtermwidget = callPackage ./qtermwidget/default.nix {};
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/coreterminal/default.nix:10]:
```suggestion
  qtermwidget = libsForQt5.callPackage ./qtermwidget { };
```

File: pkgs/applications/terminal-emulators/coreterminal/qtermwidget/default.nix [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{ stdenv
+, lib
+, fetchFromGitHub
+, python3Packages
+, cmake
+, qt5
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/coreterminal/qtermwidget/default.nix:6]:
```suggestion
, full
```

File: pkgs/applications/terminal-emulators/coreterminal/qtermwidget/default.nix [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+    rev = version;
+    sha256 = "sha256-aAeW5fPNBu+Z9//rvmGyLrXi4CZeo6bdY+YjtaqYs14=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+  ];
+
+  buildInputs = [
+    qt5.full
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/coreterminal/qtermwidget/default.nix:26]:
```suggestion
    full
```

File: pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix [hunk 7]
```diff
@@ -0,0 +1,36 @@
...
+    sha256 = "sha256-RywvFATA/+fDP/TR5QRWaJlDgy3EID//iVmrJcj3GXI=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+  ];
+
+  buildInputs = [
+    qt5.full
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix:25]:
```suggestion
    full
```

File: pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix [hunk 8]
```diff
@@ -0,0 +1,36 @@
+{ stdenv
+, lib
+, fetchzip
+, libnotify
+, cmake
+, ninja
+, qt5
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix:7]:
```suggestion
, full
```

File: pkgs/applications/terminal-emulators/coreterminal/default.nix [hunk 9]
```diff
@@ -0,0 +1,41 @@
...
+    sha256 = "sha256-YXs6VTem3AaK4n1DYwKP/jqNuf09Srn2THHyJJnArlc=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+  ];
+
+  buildInputs = [
+    qt5.full
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/coreterminal/default.nix:29]:
```suggestion
    full
```

File: pkgs/applications/terminal-emulators/coreterminal/default.nix [hunk 10]
```diff
@@ -0,0 +1,41 @@
+{ qt5
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/coreterminal/default.nix:1]:
```suggestion
{ full
```

**Review by SuperSandro2000 — CHANGES_REQUESTED** [review]

File: pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix [hunk 11]
```diff
@@ -0,0 +1,36 @@
...
+  ];
+
+  meta = with lib; {
+    description = "A library for bookmarking, saving recent activites, managing settings of C-Suite";
+    homepage = "https://gitlab.com/cubocore/coreapps/libcprime";
+    license = licenses.gpl3Plus;
+    maintainers = with maintainers; [ dan4ik605743 ];
+    platforms = platforms.linux;
+  };
+}
```
Comment (musfay) [inline pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix:36]: Could you move this expression to libraries section. This library also can be used for packaging other cubocore applications.

File: pkgs/applications/terminal-emulators/coreterminal/default.nix [hunk 12]
```diff
@@ -0,0 +1,37 @@
+{ full
+, lib
+, fetchzip
+, callPackage
+, cmake
+, ninja
+, lxqt
+}:
```
Comment (musfay) [inline pkgs/applications/terminal-emulators/coreterminal/default.nix:8]:
```suggestion
{ mkDerivation
, lib
, fetchFromGitLab
, cmake
, ninja
, qtbase
, qtserialport
, qtermwidget
}:
```

File: pkgs/applications/terminal-emulators/coreterminal/default.nix [hunk 13]
```diff
@@ -0,0 +1,37 @@
...
+
+mkDerivation rec {
+  pname = "coreterminal";
+  version = "4.2.0";
+
+  src = fetchzip {
+    url = "https://gitlab.com/cubocore/coreapps/${pname}/-/archive/v${version}/${pname}-v${version}.tar.gz";
+    sha256 = "sha256-YXs6VTem3AaK4n1DYwKP/jqNuf09Srn2THHyJJnArlc=";
+  };
```
Comment (musfay) [inline pkgs/applications/terminal-emulators/coreterminal/default.nix:17]:
```suggestion
src = fetchFromGitLab {
    owner = "cubocore/coreapps";
    repo = pname;
    rev = "v${version}";
    sha256 = "sha256-YXs6VTem3AaK4n1DYwKP/jqNuf09Srn2THHyJJnArlc=";
  };
```

File: pkgs/applications/terminal-emulators/coreterminal/default.nix [hunk 14]
```diff
@@ -0,0 +1,37 @@
...
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+  ];
+
+  buildInputs = [
+    full
+    lxqt.qtermwidget
```
Comment (musfay) [inline pkgs/applications/terminal-emulators/coreterminal/default.nix:26]:
```suggestion
    qtbase
    qtserialport
    qtermwidget
```

File: pkgs/top-level/all-packages.nix [hunk 15]
```diff
@@ -929,6 +929,8 @@ in
 
   cool-retro-term = libsForQt5.callPackage ../applications/terminal-emulators/cool-retro-term { };
 
+  coreterminal = libsForQt5.callPackage ../applications/terminal-emulators/coreterminal { };
+
```
Comment (musfay) [inline pkgs/top-level/all-packages.nix:933]:
```suggestion
  coreterminal = libsForQt5.callPackage ../applications/terminal-emulators/coreterminal {
    inherit(lxqt) qtermwidget;
  };

```

File: pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix [hunk 16]
```diff
@@ -0,0 +1,36 @@
...
+}:
+
+stdenv.mkDerivation rec {
+  pname = "libcprime";
+  version = "4.2.2";
+
+  src = fetchzip {
+    url = "https://gitlab.com/cubocore/${pname}/-/archive/v${version}/${pname}-v${version}.tar.gz";
+    sha256 = "sha256-RywvFATA/+fDP/TR5QRWaJlDgy3EID//iVmrJcj3GXI=";
+  };
```
Comment (musfay) [inline pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix:17]:
```suggestion
  src = fetchFromGitLab {
    owner = "cubocore";
    repo = pname;
    rev = "v${version}";
    sha256 = "sha256-RywvFATA/+fDP/TR5QRWaJlDgy3EID//iVmrJcj3GXI=";
  }
```

File: pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix [hunk 17]
```diff
@@ -0,0 +1,36 @@
+{ stdenv
+, lib
+, fetchzip
+, libnotify
+, cmake
+, ninja
+, full
+}:
+
+stdenv.mkDerivation rec {
```
Comment (musfay) [inline pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix:10]:
```suggestion
mkDerivation rec {
```

File: pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix [hunk 18]
```diff
@@ -0,0 +1,36 @@
+{ stdenv
+, lib
+, fetchzip
+, libnotify
+, cmake
+, ninja
+, full
+}:
```
Comment (musfay) [inline pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix:8]:
```suggestion
{ mkDerivation
, lib
, fetchFromGitLab
, libnotify
, cmake
, ninja
, qtbase
, qtconnectivity
}:
```

File: pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix [hunk 19]
```diff
@@ -0,0 +1,36 @@
...
+    sha256 = "sha256-RywvFATA/+fDP/TR5QRWaJlDgy3EID//iVmrJcj3GXI=";
+  };
+
+  nativeBuildInputs = [
+    cmake
+    ninja
+  ];
+
+  buildInputs = [
+    full
```
Comment (musfay) [inline pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix:25]:
```suggestion
    qtbase
    qtconnectivity
```

**Review by musfay — CHANGES_REQUESTED** [review]

File: pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix (hunk 11 above)
Comment (dan4ik605743) [reply pkgs/applications/terminal-emulators/coreterminal/libcprime/default.nix:36]: Yes of course

File: pkgs/top-level/all-packages.nix [hunk 20]
```diff
@@ -929,6 +929,10 @@ in
 
   cool-retro-term = libsForQt5.callPackage ../applications/terminal-emulators/cool-retro-term { };
 
+  coreterminal = libsForQt5.callPackage ../applications/terminal-emulators/coreterminal {
+    inherit(lxqt) qtermwidget;
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:933]:
```suggestion
    inherit (lxqt) qtermwidget;
```

**Review by SuperSandro2000 — COMMENTED** [review]
please squash the fixup commits

File: pkgs/applications/terminal-emulators/coreterminal/default.nix [hunk 21]
```diff
@@ -0,0 +1,41 @@
+{ mkDerivation
+, lib
+, fetchFromGitLab
+, cmake
+, ninja
+, qtbase
+, qtserialport
+, qtermwidget
```
Comment (SuperSandro2000) [inline pkgs/applications/terminal-emulators/coreterminal/default.nix:8]:
```suggestion
, qtermwidget
, libcprime
```

**Comment by dan4ik605743** [comment]
> please squash the fixup commits

Release PR with all changes?

**Comment by dan4ik605743** [comment]
> please squash the fixup commits

ok

**Comment by dan4ik605743** [comment]
@SuperSandro2000 Ready!

**Comment by dan4ik605743** [comment]
Oh shit. I did not do it ... damn. Okay, I'm already rearranged PR. Sorry.

**Pull request closed without merging**